src/Orchard/Security/Providers/FormsAuthenticationService.cs

```diff
@@ -178,5 +186,36 @@ public class FormsAuthenticationService : IAuthenticationService {
 
             return cookiePath;
         }
+
+        #region Serialization of UserData Dictionary
```

Comment:
I wrote this serialization/deserialization to avoid introducing new dependencies in Orchard.Framework.

Reply:
Json ?

Reply:
Don't even try, otherwise there is a risk that user data conflict with the format if you are not escaping correctly.

Reply:
The way it's written there should be no escaping concerns.
To safely serialize to json, I would feel safer to use JsonSerializer, or Newtonsoft.Json, or something like that. I haven't done that directly to avoid introducing new dependencies in Orchard.Framework. However, if you tell me that is fine, I would go that route.

Reply:
I think there is already a dependency on it.

Reply:
> there should be no escaping concerns

Until there is

Reply:
Ok. I will use Newtonsoft.Json